### PR TITLE
nftables support

### PR DESCRIPTION
**Pull request by Oats87**
Add iptables-detect scripts
Revert iptables to 1.8.3
Move iptables and related binaries/links to bin/aux instead of bin in order to help facilitate preferring iptables on the host rather than iptables from `k3s-root`
Create a second tar ball that only contains xtables/iptables binaries and links

https://github.com/rancher/k3s/issues/1812

### CODE REVIEW

File: iptables-detect/iptables-detect.sh [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        lsb_dist="$(. /etc/os-release && echo "$ID")"
+    fi
+
+    lsb_dist="$(echo "$lsb_dist" | tr '[:upper:]' '[:lower:]')"
+
+    # Special case redhatenterpriseserver
+    if [ "${lsb_dist}" = "redhatenterpriseserver" ]; then
+        # Set it to redhat, it will be changed to centos below anyways
+        lsb_dist='redhat'
+    fi
```
Comment (dweomer) [inline iptables-detect/iptables-detect.sh:143]: Does the `centos | redhat*)` case catch to much without this alias?

File: iptables-detect/iptables-detect.sh [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        if [ "$dist_version" -ge 8 ]; then
+            detected_via=os
+            mode=nft
+        else
+            detected_via=os
+            mode=legacy
+        fi
+        ;;
+
+        # We are running an operating system we don't know, default to nf_tables.
```
Comment (dweomer) [inline iptables-detect/iptables-detect.sh:199]: Not to make this more complicated than it needs to be, but is it worth it when encountering an unknown OS $ID to fallback to $ID_LIKE if it is present in `/etc/os-release`? Likely such is more appropriate as a future improvement.

Comment (dweomer) [reply iptables-detect/iptables-detect.sh:199]: As an example, k3OS `/etc/os-release`:
```shell
NAME="k3OS"
VERSION="k3OS v0.10.2"
ID=k3os
ID_LIKE=alpine
PRETTY_NAME="k3OS v0.10.2"
VERSION_ID="v0.10.2"
HOME_URL="https://k3os.io/"
SUPPORT_URL="https://k3os.io/"
BUG_REPORT_URL="https://github.com/rancher/k3os/issues"
ISO_URL="https://github.com/rancher/k3os/releases/download/v0.10.2/k3os-amd64.iso"
```

File: iptables-detect/iptables-detect.sh [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    *)
+        detected_via=os
+        mode=nft
+        ;;
+
+    esac
+
+}
+
+if [ ! -z "$IPTABLES_MODE" ]; then
```
Comment (dweomer) [inline iptables-detect/iptables-detect.sh:209]:
```suggestion
if [ -n "$IPTABLES_MODE" ]; then
```

File: iptables-detect/iptables-detect.sh [hunk 4]
```diff
@@ -0,0 +1,242 @@
+#!/bin/sh
+
+# Copyright 2019 The Kubernetes Authors.
+# Copyright 2020 Rancher Labs
```
Comment (dweomer) [inline iptables-detect/iptables-detect.sh:4]: Which parts of this are our contributions?

File: package/iptables/iptables.mk [hunk 5]
```diff
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-IPTABLES_VERSION = 1.8.4
+IPTABLES_VERSION = 1.8.3
```
Comment (dweomer) [inline package/iptables/iptables.mk:7]: Given that we are intentionally downgrading iptables does it make sense to leave a comment here as to why? Link to a known issue maybe?

**Comment by cjellick** [comment]
No need to separate into two prs, but i agree this is good advice for the future.

**Comment by c-x-berger** [comment]
Now that this is merged, will https://rancher.com/docs/k3s/latest/en/advanced/#enabling-legacy-iptables-on-raspbian-buster become irrelevant in the next release?

**Comment by Oats87** [comment]
@c-x-berger There is a little bit of further work to be done (namely on k3s and within the install script) but yes, in a future release (with these changes incorporated) Debian buster will be able to operate in either mode. With the changes that are implemented here, K3s will use whatever mode you have chosen with `update-alternatives` as it's going to be defaulting to the system level binaries rather than `k3s-root` binaries from the get-go.

**Comment by Oats87** [comment]
Also resolves: https://github.com/rancher/k3s/issues/1959

**Pull request merged**